app/models/manageiq/providers/vmware/cloud_manager/refresher.rb

```diff
@@ -0,0 +1,52 @@
+class ManageIQ::Providers::Vmware::CloudManager::Refresher < ManageIQ::Providers::BaseManager::Refresher
+  include ::EmsRefresh::Refreshers::EmsRefresherMixin
+
+  def collect_inventory_for_targets(ems, targets)
```

Comment:
I think you'll want to stick with legacy inventory collection for now, it will simplify all this down to just
```
def parse_legacy_inventory(ems)
  ManageIQ::Providers::Vmware::CloudManager::RefreshParser.ems_inv_to_hashes(ems)
end
```
Then all the logic you do to build up the orgs/vdcs/vapps/vms can be done in the `RefreshParser`